dp_consistency/tests/report/test_report.py

```diff
@@ -3,114 +3,176 @@
+                            Measurement(0.00, 1),
+                            Measurement(1.65, 1),
+                            Measurement(1.65, 1),
+                        ],
+                        frozenset({EDP_ONE}): [
+                            Measurement(0.00, 1),
+                            Measurement(1.65, 1),
+                            Measurement(1.65, 1),
+                        ],
+                    }
+                )
+            },
+            metric_subsets_by_parent={},
+            cumulative_inconsistency_allowed_edp_combs={},
+        )
+
+        self.__assertReportsAlmostEqual(expected, corrected, corrected.to_array())
 
     def test_can_correct_related_metrics(self):
-        ami = 'ami'
-        mrc = 'mrc'
+        ami = "ami"
+        mrc = "mrc"
         report = Report(
-            {ami: MetricReport(
-                total_campaign_reach_time_series=[Measurement(51, 1)],
-                reach_time_series_by_edp=[
-                    [Measurement(50, 1)]]),
-
+            metric_reports={
+                ami: MetricReport(
+                    reach_time_series_by_edp_combination={
+                        frozenset({EDP_ONE, EDP_TWO}): [Measurement(51, 1)],
+                        frozenset({EDP_ONE}): [Measurement(50, 1)],
+                    }
+                ),
                 mrc: MetricReport(
-                    total_campaign_reach_time_series=[Measurement(52, 1)],
-                    reach_time_series_by_edp=[
-                        [Measurement(51, 1)]])},
+                    reach_time_series_by_edp_combination={
+                        frozenset({EDP_ONE, EDP_TWO}): [Measurement(52, 1)],
+                        frozenset({EDP_ONE}): [Measurement(51, 1)],
+                    }
+                ),
+            },
             # AMI is a parent of MRC
-            {ami: [mrc]})
+            metric_subsets_by_parent={ami: [mrc]},
+            cumulative_inconsistency_allowed_edp_combs={},
+        )
 
         corrected = report.get_corrected_report()
 
-        expected = Report(
-            {ami: MetricReport(
-                total_campaign_reach_time_series=[Measurement(51, 1)],
-                reach_time_series_by_edp=[
-                    [Measurement(51, 1)]]),
 
+        expected = Report(
+            metric_reports={
+                ami: MetricReport(
+                    reach_time_series_by_edp_combination={
+                        frozenset({EDP_ONE, EDP_TWO}): [Measurement(51.5, 1)],
+                        frozenset({EDP_ONE}): [Measurement(50.5, 1)],
+                    }
+                ),
                 mrc: MetricReport(
-                    total_campaign_reach_time_series=[Measurement(51, 1)],
-                    reach_time_series_by_edp=[
-                        [Measurement(51, 1)]])},
+                    reach_time_series_by_edp_combination={
+                        frozenset({EDP_ONE, EDP_TWO}): [Measurement(51.5, 1)],
```

Comment:
Can you please double check the numbers in this test. I had to change them, I think with this new approach optimizer took a different route but the results are still valid.

Reply:
Old test case has higher squared error I think, so this is good.